crates/astria-grpc-mock-test/tests/health/test_response.rs

```diff
@@ -105,33 +100,33 @@ fn dynamic_responder(request: &HealthCheckRequest) -> HealthCheckResponse {
 #[tokio::test]
 async fn response_delay_works_as_expected() {
     let server = start_mock_server().await;
-    let mut client = HealthClient::connect(format!("http://{}", server.local_addr))
+    let mut err_client = HealthClient::connect(format!("http://{}", server.local_addr))
+        .await
+        .unwrap();
+    let mut ok_client = HealthClient::connect(format!("http://{}", server.local_addr))
         .await
         .unwrap();
     let mock = Mock::for_rpc_given("check", matcher::message_type::<HealthCheckRequest>())
         .respond_with(
             response::default_response::<HealthCheckResponse>()
                 .set_delay(Duration::from_millis(250)),
         );
-    let mock_guard = server.mocked.register_as_scoped(mock).await;
-    let rsp_fut = client.check(HealthCheckRequest {
-        name: "helloworld".to_string(),
+    mock.mount(&server.mocked).await;
+
+    let rsp_fut_expect_err = err_client.check(HealthCheckRequest {
+        service: "helloworld".to_string(),
+    });
+    let rsp_fut_expect_ok = ok_client.check(HealthCheckRequest {
         service: "helloworld".to_string(),
     });
 
-    timeout(
-        Duration::from_millis(250),
-        join(mock_guard.wait_until_satisfied(), rsp_fut),
-    )
-    .await
-    .unwrap_err();
-
-    let rsp = client
-        .check(HealthCheckRequest {
-            name: "helloworld".to_string(),
-            service: "helloworld".to_string(),
-        })
+    timeout(Duration::from_millis(200), rsp_fut_expect_err)
         .await
-        .unwrap();
-    assert_eq!(&HealthCheckResponse::default(), rsp.get_ref());
+        .unwrap_err(); // should be error
```

Comment:
I'd remove the comment as it's just noise IMO (`.unwrap_err()` is effectively saying the same thing as the comment)

Same point for the comment after `unwrap()` below.